kernel/src/scan/state.rs

```diff
@@ -138,12 +141,14 @@ pub type ScanCallback<T> = fn(
 pub fn visit_scan_files<T>(
     data: &dyn EngineData,
     selection_vector: &[bool],
+    transforms: &Vec<Option<ExpressionRef>>,
```

Comment:
any reason for `&Vec<Option<ExpressionRef>>` instead of `&[Option<ExpresisonRef>]`?

Reply:
nope, good catch. usually clippy gets these, but it didn't this time :)